analyzers/src/SonarAnalyzer.Common/Rules/ParametersCorrectOrderBase.cs

```diff
@@ -18,141 +18,161 @@
+    protected const string DiagnosticId = "S2234";
+    protected override string MessageFormat => "Parameters to '{0}' have the same names but not the same order as the method arguments.";
 
-        protected abstract TypeInfo GetArgumentTypeSymbolInfo(TArgumentSyntax argument, SemanticModel semanticModel);
-        protected abstract Location GetMethodDeclarationIdentifierLocation(SyntaxNode syntaxNode);
-        protected abstract SyntaxToken? GetArgumentIdentifier(TArgumentSyntax argument);
-        protected abstract SyntaxToken? GetNameColonArgumentIdentifier(TArgumentSyntax argument);
+    private readonly DiagnosticDescriptor rule;
 
-        internal void ReportIncorrectlyOrderedParameters(SonarSyntaxNodeReportingContext analysisContext,
-            MethodParameterLookupBase<TArgumentSyntax> methodParameterLookup, SeparatedSyntaxList<TArgumentSyntax> argumentList,
-            Func<Location> getLocationToReport)
-        {
-            var argumentParameterMappings = methodParameterLookup.GetAllArgumentParameterMappings()
-                .ToDictionary(pair => pair.Node, pair => pair.Symbol);
+    protected abstract TypeInfo GetArgumentTypeSymbolInfo(TArgumentSyntax argument, SemanticModel model);
+    protected abstract Location GetMethodDeclarationIdentifierLocation(SyntaxNode syntaxNode);
+    protected abstract SyntaxToken? GetArgumentIdentifier(TArgumentSyntax argument, SemanticModel model);
+    protected abstract SyntaxToken? GetNameColonArgumentIdentifier(TArgumentSyntax argument);
 
-            var methodSymbol = methodParameterLookup.MethodSymbol;
-            if (methodSymbol == null)
-            {
-                return;
-            }
+    protected ParametersCorrectOrderBase() : base(DiagnosticId) =>
+        rule = Language.CreateDescriptor(DiagnosticId, MessageFormat);
 
-            var parameterNames = argumentParameterMappings.Values
-                .Select(symbol => symbol.Name)
-                .Distinct()
-                .ToList();
-
-            var argumentIdentifiers = argumentList
-                .Select(argument => ConvertToArgumentIdentifier(argument))
-                .ToList();
-            var identifierNames = argumentIdentifiers
-                .Select(p => p.IdentifierName)
-                .ToList();
-
-            if (parameterNames.Intersect(identifierNames).Any() &&
-                HasIncorrectlyOrderedParameters(argumentIdentifiers, argumentParameterMappings, parameterNames, identifierNames,
-                    analysisContext.SemanticModel))
-            {
-                // for VB the symbol does not contain the method syntax reference
-                var secondaryLocations = methodSymbol.DeclaringSyntaxReferences
-                    .Select(s => GetMethodDeclarationIdentifierLocation(s.GetSyntax()))
-                    .WhereNotNull();
+    internal void ReportIncorrectlyOrderedParameters(SonarSyntaxNodeReportingContext analysisContext,
```

Comment:
Why is this internal? `protected` should be enough

Reply:
Because of [CS0051](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0051): MethodParameterLookupBase parameter is less accessible than ReportIncorrectlyOrderedParameters.